### PR TITLE
fix: use outer join to get other users

### PR DESCRIPTION
**Pull request by brm-stnd**
di querynya saya tidak menambahkan default 0 pada field follower_count karena by default postgree sudah menambahkan 0 ketika countnya null

### CODE REVIEW

**Comment by brm-stnd** [comment]
> Old migration should not be edited anymore, since it will change the integrity of the DB

fixed

File: database/migrations/2023_09_18_154312_create_vwm_user_follower_count_v2.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    public function up()
+    {
+        DB::statement("
+            CREATE MATERIALIZED VIEW vwm_user_follower_count
+            AS 
+            SELECT
+                B.user_id as user_id_followed,
+                COUNT(A.user_id_follower) as follower_count
+            FROM user_follow_user A
+            FULL OUTER JOIN users B
```
Comment (fajarism) [inline database/migrations/2023_09_18_154312_create_vwm_user_follower_count_v2.php:24]: Use right outer join
If use full outer join, there will be null value on the left

Also please filter on user that is not anonymous

**Review by fajarism — CHANGES_REQUESTED** [review]
Migration is swapped each other

**Review by fajarism — CHANGES_REQUESTED** [review]

File: database/migrations/2023_09_18_175818_create_vwm_user_topic_follower_count_v2.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                A.topic_id,
+                B.user_id,
+                C.follower_count
+            ) AS user_topic_by_follower_count
+        WHERE 
+            user_topic_by_follower_count.topic_follower_rank <= 5
+        ORDER BY 
+            user_topic_by_follower_count.topic_id ASC,
+            user_topic_by_follower_count.topic_follower_rank ASC
+        WITH DATA
```
Comment (fajarism) [inline database/migrations/2023_09_18_175818_create_vwm_user_topic_follower_count_v2.php:50]: Old migration
Use 2023_09_14_142046_create_vwm_user_topic_follower_count_rank_v5
`
CREATE MATERIALIZED VIEW vwm_user_topic_follower_count_rank
        AS
        SELECT
            *
        FROM (
            SELECT
                A.topic_id,
                C.follower_count as follower_count,
                ROW_NUMBER() OVER(
                    PARTITION BY A.topic_id
                    ORDER BY C.follower_count DESC
                ) AS topic_follower_rank,
                B.*
            FROM
                vwm_user_follower_count C
            INNER JOIN
                users B
            ON C.user_id_followed = B.user_id
            INNER JOIN
                user_topics A
            ON
                C.user_id_followed = A.user_id
						WHERE B.profile_pic_path NOT LIKE '%default-profile-picture%'
            GROUP BY
                A.topic_id,
                B.user_id,
                C.follower_count
            ) AS user_topic_by_follower_count
        WHERE
            user_topic_by_follower_count.topic_follower_rank <= 11
        ORDER BY
            user_topic_by_follower_count.topic_id ASC,
            user_topic_by_follower_count.topic_follower_rank ASC
        WITH DATA"
`

**Review by fajarism — CHANGES_REQUESTED** [review]

**Review by fajarism — CHANGES_REQUESTED** [review]
2023_09_18_175818_create_vwm_user_topic_follower_count_v2.php is still using old migration

**Review by fajarism — APPROVED** [review]

**Comment by fajarism** [comment]
Good job mas @brm-stnd

**Pull request merged**